remoteappmanager/db/orm.py

```diff
@@ -129,6 +130,54 @@ def reset(self):
         Base.metadata.create_all(self.engine)
 
 
+class AccountingInterface(ABCAccounting):
+
+    def __init__(self, url, **kwargs):
+        self.db = Database(url, **kwargs)
+
+        # We keep the same session for all transactions
+        # so that the session remains on one thread
+        self.session = self.db.create_session()
+
+    def get_user_by_name(self, user_name):
+        """ Return an orm.User given a user name.  Return None
+        if the user name is not found in the database
+        """
+        with transaction(self.session):
```

Comment:
Not now, but we will have to consider subtransactions sooner or later... This will isolate the transaction in a different session, with potential problems if another session/transaction is assuming some things.

Note that being readonly does not protect us. We can still have race conditions because we lose isolation if we terminate the transaction and resume it later. 

This is a general problem in our design though, so I am not too concerned at the moment, but it needs a bit more care in the future iterations.